### PR TITLE
docs: Add Get started docs for 1.2

### PR DESCRIPTION
**Pull request by Sherlock113**
## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [x] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: docs/source/get-started/installation.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    serve-grpc          Start a gRPC BentoServer from a given 🍱
+
+Install BentoML from source code
+--------------------------------
+
+If you want to install BentoML from the source, run the following command:
+
+.. code-block:: bash
+
+    pip install git+https://github.com/bentoml/Bentoml
```
Comment (frostming) [inline docs/source/get-started/installation.rst:72]:
```suggestion
    pip install git+https://github.com/bentoml/BentoML
```
Although case-insensitive, let's use the canonical form.

File: docs/source/get-started/installation.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+This command installs the bleeding edge ``main`` version of BentoML, which is useful for staying up-to-date with the latest features and bug fixes. However, the ``main`` version may not always be stable. If you run into any issues, please either create `an issue <https://github.com/bentoml/BentoML/issues/new/choose>`_ or join our community on `Slack <https://l.bentoml.com/join-slack>`_ to get help.
+
+Editable install
+----------------
```
Comment (frostming) [inline docs/source/get-started/installation.rst:77]: IMO this should belong to the development guide.

File: docs/source/get-started/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+In the ``Summarization`` class, the Service retrieves a pre-trained model (``sshleifer/distilbart-cnn-12-6``) from the Hugging Face hub and initializes a pipeline for text summarization. The ``summarize`` method serves as the API endpoint. In this example, it accepts a string input with an example provided, processes it through the pipeline, and returns the summarized text.
+
+Run ``bentoml serve service:<service_class_name>`` in your project directory to start the BentoML server.
+
+.. code-block:: bash
+
+    $ bentoml serve service:Summarization
+
+    2023-12-18T06:51:51+0000 [INFO] [cli] Prometheus metrics for HTTP BentoServer from "service:Summarization" can be accessed at http://localhost:3000/metrics.
+    2023-12-18T06:51:51+0000 [INFO] [cli] Starting production HTTP BentoServer from "service:Summarization" listening on http://0.0.0.0:3000 (Press CTRL+C to quit)
```
Comment (frostming) [inline docs/source/get-started/quickstart.rst:77]:
```suggestion
    2023-12-18T06:51:51+0000 [INFO] [cli] Prometheus metrics for HTTP BentoServer from "service:Summarization" can be accessed at http://localhost:3000/metrics.
    2023-12-18T06:51:51+0000 [INFO] [cli] Starting production HTTP BentoServer from "service:Summarization" listening on http://localhost:3000 (Press CTRL+C to quit)
```

File: docs/source/get-started/quickstart.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+The server is active at `http://0.0.0.0:3000 <http://0.0.0.0:3000>`_. You can interact with it in different ways.
```
Comment (frostming) [inline docs/source/get-started/quickstart.rst:79]:
```suggestion
The server is active at `http://localhost:3000 <http://localhost:3000>`_. You can interact with it in different ways.
```

File: docs/source/get-started/installation.rst (hunk 2 above)
Comment (Sherlock113) [reply docs/source/get-started/installation.rst:77]: I added a link to the Developer Guide. We can give basic explanations here and direct readers to the Developer Guide.

**Review by frostming — APPROVED** [review]

**Pull request merged**